### PR TITLE
[Firewall] bump to 0.4.0

### PR DESCRIPTION
**Pull request by haroldrandom**
1. https://github.com/Azure/azure-cli-extensions/pull/1795
2. https://github.com/Azure/azure-cli/issues/13638

---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally?

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your PR is merged into master branch, a new PR will be created to update `src/index.json` automatically.  
The precondition is to put your code inside this repo and upgrade the version in the PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Firewall

File: src/azure-firewall/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,12 @@
 Release History
 ===============
 
+
+0.4.0
+++++++
+* `az network firewall create/update`: add --dns-servers, --enable-dns-proxy, --dns-require-proxy-for-network-rules to configure DNS proxy settings.
```
Comment (haroldrandom) [inline src/azure-firewall/HISTORY.rst:9]:
```suggestion
* `az network firewall create/update`: add --dns-servers, --enable-dns-proxy, --require-dns-proxy-for-network-rules to configure DNS proxy settings.
```

File: src/azure-firewall/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,12 @@
...
+* `az network firewall policy create/update`: add --dns-servers, --enable-dns-proxy, --dns-require-proxy-for-network-rules to configure DNS proxy settings.
```
Comment (haroldrandom) [inline src/azure-firewall/HISTORY.rst:10]:
```suggestion
* `az network firewall policy create/update`: add --dns-servers, --enable-dns-proxy, --require-dns-proxy-for-network-rules to configure DNS proxy settings.
```

File: src/azure-firewall/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,12 @@
 Release History
 ===============
 
+
+0.4.0
+++++++
+* `az network firewall create/update`: add --dns-servers, --enable-dns-proxy, --require-dns-proxy-for-network-rules to configure DNS proxy settings.
+* `az network firewall policy create/update`: add --dns-servers, --enable-dns-proxy, --require-dns-proxy-for-network-rules to configure DNS proxy settings.
+
```
Comment (MyronFanQiu) [inline src/azure-firewall/HISTORY.rst:11]:
```suggestion
* `az network firewall policy create`: support threat white list arguments.
* `az network firewall create/update`: support --threat-intel-model argument
```

File: src/azure-firewall/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,13 @@
 Release History
 ===============
 
+
+0.4.0
+++++++
+* `az network firewall create/update`: add --dns-servers, --enable-dns-proxy, --require-dns-proxy-for-network-rules to configure DNS proxy settings.
+* `az network firewall policy create/update`: add --dns-servers, --enable-dns-proxy, --require-dns-proxy-for-network-rules to configure DNS proxy settings.
+* `az network firewall policy create`: support threat white list arguments.
+* `az network firewall create/update`: support --threat-intel-model argument
```
Comment (MyronFanQiu) [inline src/azure-firewall/HISTORY.rst:12]:
```suggestion
* `az network firewall create/update`: support --threat-intel-model argument.
```

**Pull request merged**